Gems/Atom/RHI/DX12/Code/Source/RHI/AsyncUploadQueue.cpp

```diff
@@ -243,232 +241,249 @@ namespace AZ
+                        uint32_t stagingSlicePitch = RHI::AlignUp(subresourceLayout.m_rowCount*stagingRowPitch, D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT);
+                        const uint32_t compressedTexelBlockSizeHeight = subresourceLayout.m_blockElementHeight;
+
+                        // ImageHeight must be bigger than or equal to the Image's row count. Images with a RowCount that is less than the ImageHeight indicates a block compression.
+                        // Images with a RowCount which is higher than the ImageHeight indicates a planar image, which is not supported for streaming images.
+                        if (subresourceLayout.m_size.m_height < subresourceLayout.m_rowCount)
                         {
-                            AZ_Warning("RHI::DX12", false, "AsyncUploadQueue staging buffer (%dK) is not big enough"
-                                "for the size of one row of image's sub-resource (%dK). Please increase staging buffer size.",
-                                m_descriptor.m_stagingSizeInBytes / 1024.0f, stagingSlicePitch / 1024.f);
-                            continue;
+                            AZ_Error("StreamingImage", false, "AsyncUploadQueue::QueueUpload expects ImageHeight '%d' to be bigger than or equal to the image's RowCount '%d'.", subresourceLayout.m_size.m_height, subresourceLayout.m_rowCount);
+                            return 0;
                         }
 
-                        needSplitSlice = true;
-                        rowsPerSplit = static_cast<uint32_t>(RHI::AlignDown(m_descriptor.m_stagingSizeInBytes, D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT) / stagingRowPitch);
-                        stagingSize = RHI::AlignUp(rowsPerSplit * stagingRowPitch, D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT);
-                        AZ_Assert(stagingSize <= m_descriptor.m_stagingSizeInBytes, "Final staging size can't be larger than staging buffer size");
-                    }
+                        // The final staging size for each CopyTextureRegion command
+                        uint32_t stagingSize = stagingSlicePitch;
 
-                    if (!needSplitSlice)
-                    {
-                        // Try to use one frame packet for all sub-resources if it's possible.
-                        for (const auto& subresource : request.m_mipSlices[sliceIndex].m_subresources)
+                        // Prepare for spliting this subresource if needed
+                        if (stagingSlicePitch > m_descriptor.m_stagingSizeInBytes)
                         {
-                            for (uint32_t depth = 0; depth < subresourceLayout.m_size.m_depth; depth++)
+                            // Calculate minimum size of one row of this subresrouce
+                            uint32_t minSize = RHI::AlignUp(stagingRowPitch, D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT);
```

Comment:
I think it's using D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT because that's the upload buffer's alignment. The line following it is just to make sure the staging buffer can update at least one row pitch. 

Btw, the stagingRowPitch is already aligned with D3D12_TEXTURE_DATA_PITCH_ALIGNMENT at line 290.

Reply:
BTW, the implementation of QueueUpload(StreamingImageExpandRequest wasn't updated much except the logic to cache the request is added in this PR.